src/pydvl/influence/torch/util.py

```diff
@@ -598,3 +608,141 @@ def __init__(self, original_exception: RuntimeError):
+
+    Attributes:
+        model: The neural network model.
+        detach: Whether to detach the parameters from the computation graph.
+    """
+
+    model: torch.nn.Module
+    detach: bool = True
+
+    def _optional_detach(self, p: torch.nn.Parameter):
+        if self.detach:
+            return p.detach()
+        return p
+
+    def build(self, block_structure: OrderedDict[str, List[str]]):
+        """
+        Builds an ordered dictionary of model parameters based on the specified block
+        structure represented by an ordered dictionary, where the keys are block
+        identifiers and the values are lists of model parameter names contained in
+        this block.
+
+        Args:
+            block_structure: The block structure specifying how to group the parameters.
+
+        Returns:
+            An ordered dictionary of ordered dictionaries, where the outer dictionary's
+            keys are block identifiers and the inner dictionaries map parameter names
+            to parameters.
+        """
+        parameter_dict = {}
+
+        for block_name, parameter_names in block_structure.items():
+            inner_ordered_dict = {}
+            for parameter_name in parameter_names:
+                parameter = self.model.state_dict()[parameter_name]
+                if parameter.requires_grad:
+                    inner_ordered_dict[parameter_name] = self._optional_detach(
+                        parameter
+                    )
+                else:
+                    warnings.warn(
+                        f"The parameter {parameter_name} from the block "
+                        f"{block_name} is mark as not trainable in the model "
+                        f"and will be excluded from the computation."
+                    )
+            parameter_dict[block_name] = inner_ordered_dict
+
+        return parameter_dict
+
+    def build_from_block_mode(
+        self, block_mode: BlockMode
+    ) -> Dict[str, Dict[str, torch.nn.Parameter]]:
+        """
+        Builds an ordered dictionary of model parameters based on the specified block
+        mode or custom blocking structure represented by an ordered dictionary, where
+        the keys are block identifiers and the values are lists of model parameter names
+        contained in this block.
+
+        Args:
+            block_mode: The block mode specifying how to group the parameters.
+
+        Returns:
+            An ordered dictionary of ordered dictionaries, where the outer dictionary's
+            keys are block identifiers and the inner dictionaries map parameter names
+            to parameters.
+        """
+        parameter_dict = {}
+
+        if block_mode is BlockMode.FULL:
+            inner_ordered_dict = {}
+            for k, v in self.model.named_parameters():
+                if v.requires_grad:
+                    inner_ordered_dict[k] = self._optional_detach(v)
+            parameter_dict[""] = inner_ordered_dict
```

Comment:
I would prefer a more explicit name over an empty string here.

Reply:
Empty string for as identifier for the full model is a pytorch specific.